perun-core/src/main/java/cz/metacentrum/perun/core/entry/ResourcesManagerEntry.java

```diff
@@ -1037,6 +1037,28 @@ public void copyGroups(PerunSession sess, Resource sourceResource, Resource dest
 		}
 	}
 
+	@Override
+	public List<Resource> getResourcesWhereUserIsAdmin(PerunSession sess, Vo vo, User authorizedUser) throws InternalErrorException, PrivilegeException, UserNotExistsException, VoNotExistsException {
+		Utils.checkPerunSession(sess);
+		getPerunBl().getVosManagerBl().checkVoExists(sess, vo);
+		getPerunBl().getUsersManagerBl().checkUserExists(sess, authorizedUser);
+
+		List<Resource> resources = getResourcesManagerBl().getResourcesWhereUserIsAdmin(sess, vo, authorizedUser);
+
+		//Vo manager of the vo and perunobserver can see all returned resources
+		if(AuthzResolver.isAuthorized(sess, Role.VOADMIN, vo) ||
+			AuthzResolver.isAuthorized(sess, Role.PERUNOBSERVER)){
+			return resources;
+		}
+
+		//Resource manager can see only resources where he has role resource manager (filter them)
+		if(AuthzResolver.isAuthorized(sess, Role.RESOURCEADMIN)) {
+			return filterNotAuthorizedResource(sess, resources);
+		} else {
+			throw new PrivilegeException(sess, "getResourcesByResourceManager");
```

Comment:
This String should correspond to the name of the called method.
```suggestion
			throw new PrivilegeException(sess, "getResourcesWhereUserIsAdmin");
```